modules/ROOT/pages/configuration/configuration-settings.adoc

```diff
@@ -1952,6 +1952,32 @@ m|+++false+++
 |===
 
 
+[[config_dbms.cypher.infer_schema_parts]]
+=== `dbms.cypher.infer_schema_parts`
+
+.dbms.cypher.infer_schema_parts
+[frame="topbot", stripes=odd, grid="cols", cols="<1s,<4"]
+|===
+|Description
+a|For some queries, the planner can infer predicates such as labels or types from the graph structure that can improve estimating the number of rows that each operator produces.
+(See link:{neo4j-docs-base-uri}/cypher-manual/{page-version}/planning-and-tuning/execution-plans/#runtimes-reading-execution-plans[Cypher Manual -> Execution plans and query tuning -> Understanding execution plans])
+This setting controls to what extent it should do that:
+
+* `OFF`: No predicates are inferred.
+* `MOST_SELECTIVE_LABEL`: Relationship types are used to infer labels on the relationships' end nodes and the label corresponding to the least number of nodes is used to estimate rows.
+Not inferring more labels yields better results for nodes with several labels which are dependent, such as every `:Actor` being a `:Person`.
+
+// In general, inferring more information should improve the estimation and thereby the planner's decisions.
```

Comment:
The comments should also be added to the codebase if we want them here; otherwise, the test will fail.

Reply:
I am not sure I follow what test would fail.

The idea behind the text in the comment was that it only makes sense to me once [we have flipped the default from `OFF` to `MOST_SELECTIVE_LABEL`.](https://trello.com/c/2IqnQCAc/2275-turn-on-label-inference-by-default)

My idea was that I could already phrase things so that the next developer could just enable the text. But I did not want to commit just yet, as toggling the feature flag might slip the release.

Reply:
We are testing all settings to make sure that all documented valid values and default values are up-to-date with the codebase. If there are mismatches, then the test fails. I am currently working on fixing the mismatches in the descriptions so we can test them as well. In other words, the documentation should reflect the code literary.

For example, the failing test below shows that there is no such configuration setting in the codebase.